packages/itwin/grouping-mapping-widget/src/components/Properties/CalculatedProperties/CalculatedPropertyAction.tsx

```diff
@@ -33,39 +33,45 @@ export const CalculatedPropertyAction = ({
   onClickCancel,
 }: CalculatedPropertyActionProps) => {
   const { getAccessToken, iModelId } = useGroupingMappingApiConfig();
-  const mappingClient = useMappingClient();
+  const propertiesClient = usePropertiesClient();
   const [propertyName, setPropertyName] = useState<string>(
     calculatedProperty?.propertyName ?? "",
   );
-  const [type, setType] = useState<CalculatedPropertyType | undefined>(calculatedProperty?.type);
+  const [type, setType] = useState<CalculatedPropertyType | undefined>(calculatedProperty?.calculatedPropertyType);
```

Comment:
`type` might be `null` as well.

`type` will be `undefined` when `calculatedProperty` is null or undefined.
`type` will be `null` when `calcualtedProperty` is valid, but the `calculatedPropertyType` is null (API can return null).

Reply:
Got it, currently the UI requires `calculatedPropertyType` to create a Calculated Property, taking `null` into account here would lead to `SharedCalculatedPropertyForms` component to consider `null`, I think at this point, this refactor will come with the redesign of the UI for the properties, it could that these property categories get removed in favor of the flat-list but I'm not sure how that would look like yet.

Reply:
Fall back to undefined if property type is `null`.

Reply:
I think it can be reduced to ```calculatedProperty?.calculatedPropertyType ?? undefined```